src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -137,247 +122,110 @@ static void smart_amp_set_params(struct comp_dev *dev,
 
 		sink_c->buffer_fmt = out_fmt.interleaving_style;
 		params->frame_fmt = sink_c->stream.frame_fmt;
-
 		sink_c->hw_params_configured = true;
-
 		buffer_release(sink_c);
 	}
 }
 
-static inline int smart_amp_set_config(struct comp_dev *dev, const char *data,
-				       uint32_t data_size)
-{
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
-	struct sof_smart_amp_config *cfg;
-	uint32_t cfg_size;
-
-	cfg = (struct sof_smart_amp_config *)data;
-	cfg_size = data_size;
-
-	if (cfg_size != sizeof(struct sof_smart_amp_config)) {
-		comp_err(dev, "smart_amp_set_config(): invalid config size %u, expect %u",
-			 cfg_size, sizeof(struct sof_smart_amp_config));
-		return -EINVAL;
-	}
-
-	comp_dbg(dev, "smart_amp_set_config(): config size = %u", cfg_size);
-
-	memcpy_s(&sad->config, sizeof(struct sof_smart_amp_config), cfg,
-		 sizeof(struct sof_smart_amp_config));
-
-	return 0;
-}
-
-static inline int smart_amp_get_config(struct comp_dev *dev, char *data,
-				       uint32_t *data_size)
+static int smart_amp_set_config(struct processing_module *mod, uint32_t config_id,
+				enum module_cfg_fragment_position pos, uint32_t data_offset_size,
+				const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+				size_t response_size)
 {
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
-	uint32_t cfg_size;
+	struct comp_dev *dev = mod->dev;
+	struct smart_amp_data *sad = module_get_private_data(mod);
 
-	cfg_size = sizeof(struct sof_smart_amp_config);
-
-	if (cfg_size > *data_size) {
-		comp_err(dev, "smart_amp_get_config(): wrong config size %d",
-			 *data_size);
-	}
+	comp_info(dev, "smart_amp_set_config()");
```

Comment:
comp_dbg

Reply:
I used to, but this would lead to internal CI test failed.